Source/Dafny/Compiler-js.cs

```diff
@@ -839,19 +849,19 @@ protected class ClassWriter : IClassWriter {
         if (ComplicatedTypeParameterForCompilation(argType)) {
           Error(tok, "compilation of set<TRAIT> is not supported; consider introducing a ghost", wr);
         }
-        return DafnySetClass + "<" + TypeName(argType, wr, tok) + ">";
+        return erased ? DafnySetClass : DafnySetClass + "<" + TypeName(argType, wr, tok, erased) + ">";
```

Comment:
The angle brackets here in the JavaScript file is a latent bug. It's not legal JavaScript syntax, and such type parameters are not used in JavaScript in the first place. (In fact, it is rare that `TypeName` is called for JavaScript at all. This is why this bug has not been detected until now.) So, L852 should return just `DafnySetClass` regardless of the value of `erased`. This, in turn, means that the `erased` parameter is not needed in this file.

Reply:
Fixed